backend.native/cli.bc/src/org/jetbrains/kotlin/cli/bc/K2Native.kt

```diff
@@ -89,8 +89,10 @@ class K2Native : CLICompiler<K2NativeCompilerArguments>() {
         return ExitCode.OK
     }
 
-    val K2NativeCompilerArguments.isUsefulWithoutFreeArgs: Boolean
-        get() = listTargets || listPhases || checkDependencies || !includes.isNullOrEmpty()
+    fun isUsefulWithoutFreeArgs(arguments: K2NativeCompilerArguments, configuration: CompilerConfiguration): Boolean =
+            arguments.listTargets || arguments.listPhases || arguments.checkDependencies ||
+                    !arguments.includes.isNullOrEmpty() ||
+                    configuration.get(KonanConfigKeys.PRODUCE)!!.isCache
```

Comment:
As per our discussion, I still believe cache shalln't be usable for compilations . without free args or `-Xinclude`.